### PR TITLE
New folder structure and build process 

### PR DESCRIPTION
**Pull request by diaswrd**
_Previously discussed here:_ https://groups.google.com/forum/#!topic/foam-framework-discuss/OgUtVg7masA

**What this PR does:**
- Move everything from `java_src/` to `src/` folder, keeping the same internal folder structure
- Remove `java_src/` folder
- Update `gen.sh` to copy and generate all the files into a `build/` folder (already _gitignored_)

**An example of `sh gen.sh && git status` command before this PR:**

screen shot 2017-05-30 at 4 02 52 pm

**An example after:**

screen shot 2017-05-30 at 4 11 32 pm

**New build process:**
```
cd src/
sh gen.sh
cd ../build
mvn clean && mvn compile
```

### CODE REVIEW

File: java_src/foam/dao/index/AltIndex.java [hunk 1]
```diff
@@ -1,96 +0,0 @@
-/**
```
Comment (kgrgreer) [inline java_src/foam/dao/index/AltIndex.java]: Where did the java_src/foam/dao/index files go?

Comment (diaswrd) [reply java_src/foam/dao/index/AltIndex.java]: I assume it's generated by the `src/foam/dao/index/*.js` files? Or are they different things?

Comment (diaswrd) [reply java_src/foam/dao/index/AltIndex.java]: Just noticed that they don't have the _generated-code_ header.

So I assume `src/foam/dao/index/AltIndex.js` does not generate a `build/foam/dao/index/AltIndex.java` file?

**Comment by diaswrd** [comment]
The necessary fixes have been made. 👍

**Comment by kgrgreer** [comment]
Please resolve conflicts so we can merge. Thx

**Comment by kgrgreer** [comment]
Did you add the build instructions to the README, since people aren't going to know to read this PR for info?

**Pull request merged**